aten/src/ATen/native/cuda/ForeachReduceOp.cu

```diff
@@ -72,17 +78,27 @@ struct LpNormFunctor {
           int i = i_start + threadIdx.x + ii * blockDim.x;
           if (i < n && i < chunk_size) {
             opmath_t next = static_cast<opmath_t>(x[i]);
-            vals[ii] += NormType == 1 ? ::abs(next) : next * next;
+            if (NormType == 1234567890) {
+              vals[ii] = ::max(vals[ii], ::abs(next));
+            } else {
+              vals[ii] += NormType == 1 ? ::abs(next) : next * next;
+            }
           }
         }
       }
     }
 
     auto val = opmath_t(0);
     for (int i = 0; i < kILP; i++) {
-      val += vals[i];
+      if (NormType == 1234567890) {
```

Comment:
you should name this

Reply:
I renamed as an enum, let me know if that doesn't address your comment sufficiently!